### PR TITLE
Added 1x sentence to clarify use/support for IAM Instance Profiles.

### PR DESCRIPTION
**Pull request by etgraylog**
### Description
This adds a notation to the documentation for AWS IAM Instance profiles and their use with  OpenSearch Snapshots.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6339

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md [hunk 1]
```diff
@@ -101,6 +101,8 @@ You will most likely not need to specify any parameters except for `location`. F
 
    After the Docker cluster starts, skip to step 7.
 
+   If you're using AWS IAM Instance Profiles to allow OpenSearch nodes on AWS EC2 instances to inherit Roles for Policies granting access to AWS S3 buckets, you can skip to step 8.
```
Comment (Naarcha-AWS) [inline _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:104]:
```suggestion
   If you're using [AWS IAM instance profile](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html) to allow OpenSearch nodes on AWS EC2 instances to inherit roles for policies when granting access to AWS S3 buckets, skip to step 8.

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md (hunk 1 above)
Comment (Naarcha-AWS) [reply _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:104]: @scrawfor99, can you verify that this is technically accurate?

Comment (scrawfor99) [reply _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:104]: Hi @Naarcha-AWS, I don't have any true expertise in this area. However, judging from the existing documentation on the snapshots docs and the linked instance profile documentation this sounds reasonable. I am not able to confirm this without testing though so perhaps @etgraylog could share whether they have gone through these steps themselves (I would assume so) and that will work for verification.

Comment (etgraylog) [reply _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:104]: > Hi @Naarcha-AWS, I don't have any true expertise in this area. However, judging from the existing documentation on the snapshots docs and the linked instance profile documentation this sounds reasonable. I am not able to confirm this without testing though so perhaps @etgraylog could share whether they have gone through these steps themselves (I would assume so) and that will work for verification.

Thank you @scrawfor99! Yes exactly, this discovery is from troubleshooting a deployment that makes use of an AWS IAM Instance Profile (that grants access to the S3 bucket of focus). When storing AWS IAM Role credentials (for S3 bucket access) in the OpenSearch Keystore, which the EC2 Instance(s) already had the ability to access via IAM Instance Profile granting the AWS IAM Role, failures would occur when attempting to access the bucket. After removing them from the OpenSearch Keystore while leaving the IAM Instance Profile, etc... in-place, access to the S3 Bucket worked as expected.

And thank you @Naarcha-AWS for adding the hyperlink - good idea! 👍 I'll sign off and commit that suggestion.

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**